vm/src/obj/objcomplex.rs

```diff
@@ -308,3 +309,39 @@ struct ComplexArgs {
     #[pyarg(positional_or_keyword, default = "None")]
     imag: Option<PyObjectRef>,
 }
+
+fn parse_str(s: &str) -> Option<Complex64> {
+    // Handle parentheses
+    let s = match s.strip_prefix('(') {
+        None => s,
+        Some(s) => match s.strip_suffix(')') {
+            None => return None,
+            Some(s) => s.trim(),
+        },
+    };
+
+    let value = match s.strip_suffix(|c| c == 'j' || c == 'J') {
+        None => Complex64::new(float_ops::parse_str(s)?, 0.0),
+        Some(mut s) => {
+            let mut real = 0.0;
+            // Find the central +/- operator. If it exists, parse the real part.
+            for (i, (a, b)) in s.chars().tuple_windows().enumerate() {
```

Comment:
Because every characters are ascii character, I think same algorithm will perfectly work as bytes. Will you try same version with `s.as_bytes()` isntead of `s.chars()`?

Reply:
Fixed it! Thank you for the comment